sdks/swift/Mentat/Mentat/Query/Query.swift

```diff
@@ -0,0 +1,218 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Mentatlib
+
+class Query: OptionalRustObject {
+
+    func bind(varName: String, toLong value: Int64) throws -> Query {
+        guard let r = self.raw else {
+            throw QueryError.builderConsumed
+        }
+        query_builder_bind_long(r, varName, value)
+        return self
+    }
+
+    func bind(varName: String, toReference value: Int64) throws -> Query {
+        guard let r = self.raw else {
+            throw QueryError.builderConsumed
+        }
+        query_builder_bind_ref(r, varName, value)
+        return self
+    }
+
+    func bind(varName: String, toReference value: String) throws -> Query {
+        guard let r = self.raw else {
+            throw QueryError.builderConsumed
+        }
+        query_builder_bind_ref_kw(r, varName, value)
+        return self
+    }
+
+    func bind(varName: String, toKeyword value: String) throws -> Query {
+        guard let r = self.raw else {
+            throw QueryError.builderConsumed
+        }
+        query_builder_bind_kw(r, varName, value)
+        return self
+    }
+
+    func bind(varName: String, toBoolean value: Bool) throws -> Query {
+        guard let r = self.raw else {
+            throw QueryError.builderConsumed
+        }
+        query_builder_bind_boolean(r, varName, value ? 1 : 0)
+        return self
+    }
+
+    func bind(varName: String, toDouble value: Double) throws -> Query {
+        guard let r = self.raw else {
+            throw QueryError.builderConsumed
+        }
+        query_builder_bind_double(r, varName, value)
+        return self
+    }
+
+    func bind(varName: String, toDate value: Date) throws -> Query {
+        guard let r = self.raw else {
+            throw QueryError.builderConsumed
+        }
+        query_builder_bind_timestamp(r, varName, value.toMicroseconds())
+        return self
+    }
+
+    func bind(varName: String, toString value: String) throws -> Query {
+        guard let r = self.raw else {
+            throw QueryError.builderConsumed
+        }
+        query_builder_bind_string(r, varName, value)
+        return self
+    }
+
+    func bind(varName: String, toUuid value: UUID) throws -> Query {
+        guard let r = self.raw else {
+            throw QueryError.builderConsumed
+        }
+        query_builder_bind_uuid(r, varName, value.uuidString)
+        return self
+    }
+
+    func executeMap(map: @escaping (TupleResult?, QueryError?) -> Void) throws {
+        guard let r = self.raw else {
+            throw QueryError.builderConsumed
+        }
+
+        DispatchQueue.global(qos: .background).async {
+            let result = query_builder_execute(r)
```

Comment:
So the query builder here is capturing a `Store` instance as its hidden raw pointer? I don't think I understand how this works, because I don't see where the reference to the SQLite connection and the `Store` are managed.

Reply:
When we create a `Query` we pass in a reference to store (see `Mentat.query`). We return a pointer to the query builder which we store in `Query`. Therefore we have:
`Mentat` holding the pointer to `Store` as it's raw pointer.
`Query` holding the pointer to `QueryBuilder` as it's raw pointer.
`QueryBuilder` holding a reference to `Store` in Rust.